### DISCUSSION TITLE
GetFullList Takes Long (1.5 s) at first request

### DISCUSSION THREAD

**Discussion by milimyname** [pocketbase/pocketbase#3995] [Q&A]
Hi everyone.

I just got a sveltekit app and pocketbase. Both of them deployed on Fly.io and in the same region (ams). However, the issue is that 
the first request from `+page.server.ts` takes about 1.5 seconds. The following requests are under 40 ms

1. I've tried to fetch only 10, not the whole collection - Didn't work
2. Tried to implement the sveltekit [streaming](https://svelte.dev/blog/streaming-snapshots-sveltekit) - Didn't work
3. Try to add some options to getFullList but it didn't work. That's where I am. I guess it is more to do with servers but maybe I missing something. 

```+page.server.ts

/** @type {import('./$types').PageServerLoad} */
export const load = async ({ locals, params }) => {
	// Get only 10 flashcards at a time

	console.time('getFirstTenFlashcards');
	const firstTenFlashcards = await locals.pb.collection('flashcard').getFullList({
		filter: `flashcardBox = "${params.slug}"`,
		fields: `id, name, meaning, romanji, furigana, type, notes`,
		skipTotal: true, // i guess it is not necassary
		batch: 20 
	});

	if (!kuroshiroInitialized) {
		await kuroshiro.init(new KuromojiAnalyzer());
		kuroshiroInitialized = true;
	}

	// Process furigana in parallel
	const processeFirstTenFlashcards = await Promise.all(
		firstTenFlashcards.map((card) => processFurigana(card))
	);

	// Server API:
	const form = await superValidate(flashcardSchema);

	console.timeEnd('getFirstTenFlashcards');

	return {
		form,
		flashcards: processeFirstTenFlashcards
	};
};

async function processFurigana(card: RecordModel) {
	// Custom furigana processing
	if (card.furigana.includes('/') && isHiragana(card.furigana[card.furigana.indexOf('/') + 1])) {
		card.customFurigana = card.furigana;
		card.furigana = convertToRubyTag(card.furigana);
	} else {
		// Kuroshiro conversion for kanji cards
		card.furigana = await kuroshiro.convert(card.name, {
			to: 'hiragana',
			mode: 'furigana'
		});
	}

	return card;
}

```

If you need the `+page.svelte`, here is the [repo](https://github.com/milimyname/taijobi/tree/main/src/routes/flashcards/%5Bslug%5D)

**Comment by ganigeorgiev** [comment]
There is too much going on in the provided repo to try to pinpoint the exact cause of the slow response.

Are you sure that the `getFullList()` call is causing the slow response time and not the `KuromojiAnalyzer`, the _"Process furigana in parallel"_ or _`await superValidate(flashcardSchema)`_?

**Reply by ganigeorgiev** [reply]
Is the issue also reproduced locally or only when deployed in Fly?

Note that it is possible the Fly VM to go in a "suspend" mode if not used and the slow response to be caused by the "wake up" time.
If the issue is reproduced only on the Fly VM you can try to disable the `auto_stop_machines` and set `min_machines_running=1` (see also https://fly.io/docs/apps/autostart-stop/).

**Reply by milimyname** [reply]
I thought of it as well and it is when deployed in Fly.  I guess it ain't about the code issues. I need to let to run at least one machine.

**Reply by ganigeorgiev** [reply]
It could be also code related, I don't know. Try to also specify a more granular `console.timeEnd` (eg. right after the `getFullList` call) to target only the PocketBase call. You are measuring too much things in the above code.

You can also try specifying an index on the `flashcardBox` field if you haven't already to try optimize the query, but if your current execution takes 1.5s only the first time then the issue is probably something else (we don't have cache so it doesn't make much sense why it runs so much faster after the first request).
